packages/yarnpkg-builder/sources/tools/makeConfig.ts

```diff
@@ -83,6 +83,15 @@ export const makeConfig = (config: webpack.Configuration): webpack.Configuration
       resourceRegExp: /^encoding$/,
       contextRegExp: /node-fetch/,
     }),
+    // esprima is only needed for parsing !!js/function, which isn't part of the FAILSAFE_SCHEMA.
+    // Unfortunately, js-yaml declares it as a hard dependency and requires the entire module,
+    // which causes webpack to add 0.13 MB of unused code to the bundle.
+    // Fortunately, js-yaml wraps the require call inside a try / catch block, so we can just ignore it.
+    // Reference: https://github.com/nodeca/js-yaml/blob/master/lib/js-yaml/type/js/function.js#L15
+    new webpack.IgnorePlugin({
+      resourceRegExp: /^esprima$/,
+      contextRegExp: /js-yaml/,
+    }),
```

Comment:
This might be a slight problem for anyone using `js-yaml` in their plugin and actually needing this.

Reply:
Oh, right - I put it in the wrong place. Thanks!

Reply:
> This might be a slight problem for anyone using `js-yaml` in their plugin and actually needing this.

Aha! 😄 

@merceyz I think I've found the source of a pretty obscure bug. I think my `package.yaml` plugin is the source of the following error. It doesn't always occur, and hasn't appeared on my smaller repos.

```
(node:66415) [MODULE_NOT_FOUND] Error: Your application tried to access esprima, but it isn't declared in your dependencies; this makes the require call ambiguous and unsound.
(Use `node --trace-warnings ...` to show where the warning was created)
```

Adding `esprima` to `dependencies` on the project root `package.json` resolves the issue.

Is there a way I can just bundle it? Or indicate to yarn that the plugin is a dependency?

Reply:
After the switch to esbuild in https://github.com/yarnpkg/berry/pull/2362 and https://github.com/yarnpkg/berry/pull/2717 we no longer mark `esprima` as external, though somehow it still isn't bundled.

Reply:
I posted this in discord, and got some good direction on an approach that might bypass the issue.